app/script/view_model/content/InputBarViewModel.js

```diff
@@ -306,13 +333,16 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
       messageEntity.isEditing(true);
       this.editMessageEntity(messageEntity);
 
-      // clean mentions before setting text in order to prevent mentions offset recomputing
-      this.currentMentions.removeAll();
       this.input(messageEntity.get_first_asset().text);
+      this.currentMentions(
+        messageEntity
+          .get_first_asset()
+          .mentions()
+          .slice()
```

Comment:
Let's pull tehse into a const and then put them into the observable. This is hard to read.